### PR TITLE
[BD-18] Standardize the make test targets

### PR DESCRIPTION
**Pull request by emzubair**
## Standardize the make test targets
Added target for test & clean

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @emzubair! I've created [BLENDED-463](https://openedx.atlassian.net/browse/BLENDED-463) to keep track of it in Jira.  More details are on the [BD-18](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1593934815/BD-18+Testing+Consistency+Improvements) project page.

When this pull request is ready, tag your edX technical lead.

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,18 @@
+.PHONY: clean test upgrade quality
```
Comment (feanil) [inline Makefile:1]: Update to add all the targets added below including `requirements`

File: Makefile [hunk 2]
```diff
@@ -8,3 +23,27 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
...
+quality-js: ## Run JavaScript linter
+	echo "Tool not configured for quality target"
+
+quality-python: ## Run python linters
+	pycodestyle edx_notifications
+	pylint edx_notifications --reports=no
+
+quality: quality-js quality-python ## Run linters
+
+requirements: ## install development environment requirements
```
Comment (feanil) [inline Makefile:38]: Update `travis.yml` to use the new targets.

File: Makefile [hunk 3]
```diff
@@ -8,3 +23,27 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
 	sed -i.tmp '/^[dD]jango==/d' requirements/testing.txt
 	rm requirements/testing.txt.tmp
 
+install-js: ## install JavaScript dependencies
+	npm install
+
+quality-js: ## Run JavaScript linter
```
Comment (feanil) [inline Makefile:29]: Remove the target if it's not doing anything.

**Comment by emzubair** [comment]
@feanil updated the `PR`

File: .travis.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ envs:
 
 script:
   - tox
-  - ./node_modules/gulp/bin/gulp.js test
+  - make test-js
   - bash ./run_bokchoy_tests.sh
```
Comment (feanil) [inline .travis.yml:14]: Why isn't run_bokchoy_tests part of the `test` target?  Given that it's part of the test suite, it seems like we would want to run it locally as well.

File: tox.ini [hunk 5]
```diff
@@ -16,5 +16,4 @@ commands =
 
 [textenv:quality]
 commands =
-    pycodestyle edx_notifications
-    pylint edx_notifications --reports=no
+    make quality
```
Comment (feanil) [inline tox.ini:19]: I would flip this around the other way.  `make quality` should run `tox -e quality`

**Comment by emzubair** [comment]
@feanil all tests are covered under `test` target now. `PR` updated

**Pull request merged**